frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segment_statistics_modal.tsx

```diff
@@ -233,33 +228,38 @@ export function SegmentStatisticsModal({
     return potentialGroupNode?.name == null ? "" : potentialGroupNode.name;
   };
 
-  const isErrorCase = dataSource === ERROR_CODE;
+  const isErrorCase = segmentStatisticsObjects == null;
 
   return (
     <Modal
       open
       title="Segment Statistics"
       onCancel={onCancel}
       width={700}
-      onOk={() => exportStatisticsToCSV(dataSource, tracingId, parentGroup, hasAdditionalCoords)}
+      onOk={() =>
+        exportStatisticsToCSV(segmentStatisticsObjects, tracingId, parentGroup, hasAdditionalCoords)
+      }
       okText="Export to CSV"
       okButtonProps={{ disabled: isErrorCase }}
     >
-      <Spin spinning={dataSource?.length === 0}>
+      <Spin spinning={segmentStatisticsObjects?.length === 0 && segments.length > 0}>
         {isErrorCase ? (
           MODAL_ERROR_MESSAGE
         ) : (
           <>
             {hasAdditionalCoords && (
               <Alert
                 className="segments-stats-info-alert"
-                message={`These statistics are only valid for the current additional coordinates ${additionalCoordinateStringForModal}.`}
+                message={`These statistics only refer to the current additional ${pluralize(
+                  "coordinate",
+                  additionalCoordinates?.length || 0,
+                )} ${additionalCoordinateStringForModal}.`}
                 type="info"
                 showIcon
               />
             )}
             <Table
-              dataSource={dataSource as Array<SegmentInfo>}
+              dataSource={segmentStatisticsObjects as Array<SegmentInfo>}
```

Comment:
please try to avoid (such) casts as much as possible, since they are weakinging TS ability to warn about problems. I think, in this particular case you need to inline the value of `isErrorCase` in line 246. Then, TS should understand that that `segmentStatisticsObjects` won't be null in this if-else-branch.